kokain-ksp/src/main/java/com/schwarz/kokain/ksp/validation/PreValidator.kt

```diff
@@ -0,0 +1,80 @@
+package com.schwarz.kokain.ksp.validation
+
+import com.google.devtools.ksp.KspExperimental
+import com.google.devtools.ksp.getJavaClassByName
+import com.google.devtools.ksp.getKotlinClassByName
+import com.google.devtools.ksp.processing.KSPLogger
+import com.google.devtools.ksp.processing.Resolver
+import com.google.devtools.ksp.symbol.KSClassDeclaration
+import com.google.devtools.ksp.symbol.KSType
+import com.google.devtools.ksp.symbol.Modifier
+import com.schwarz.kokain.api.EBean
+import com.schwarz.kokain.api.EFactory
+import com.schwarz.kokain.api.KDiFactory
+import com.schwarz.kokain.ksp.util.getKSAnnotationsByType
+import com.schwarz.kokain.ksp.util.getKSValueArgumentByName
+import com.squareup.kotlinpoet.ksp.toTypeName
+
+class PreValidator(logger: KSPLogger, val resolver: Resolver) {
+
+    private val logger = logger
+
+    @Throws(ClassNotFoundException::class)
+    fun validateEbean(model: KSClassDeclaration): Boolean {
+
+        val entityElement = model
+        var result = true
+
+        if (entityElement.modifiers.contains(Modifier.PRIVATE)) {
+            logger.error(EBean::class.java.simpleName + " can not be private", entityElement)
+            result = false
+        }
+        if (entityElement.modifiers.contains(Modifier.PROTECTED)) {
+            logger.error(EBean::class.java.simpleName + " can not be protected", entityElement)
+            result = false
+        }
+        if (entityElement.modifiers.contains(Modifier.FINAL)) {
+            logger.error(EBean::class.java.simpleName + " can not be final", entityElement)
+            result = false
+        }
+
+        entityElement.primaryConstructor?.let {
+            if (it.parameters.isNotEmpty()) {
+                logger.error(EBean::class.java.simpleName + " should not have a contructor", it)
```

Comment:
maybe "should not have a constructor with arguments"?
btw: could we not allow arguments with default-values?
(sorry, missing a bit of context here, so maybe the question does not make much sense :D)

Reply:
it's not the worst question I guess. The Idea is that the GeneratedFactory is always able to create the beans which should be injected. So in general you're right we could also allow default constructor parameters but from my perspective this doesn't make sense because they're basically const values than and why don't we use for constants just constants. Or did I miss something here?

Reply:
Unless you also want to create the beans manually in some cases.
But I guess that does not make much sense because we should not that anyway 👍